src/modules/pivotTable/applyLegendSet.js

```diff
@@ -28,17 +26,17 @@ const getLegendSet = engine => {
     return engine.legendSets[legendSetId]
 }
 
-const buildStyleObject = (legend, engine) => {
+const buildStyleObject = (legendColor, engine) => {
     const style = {}
     switch (engine.visualization.legendDisplayStyle) {
         case LEGEND_DISPLAY_STYLE_TEXT:
-            style.color = legend.color
+            style.color = legendColor
             // TODO: Adapt background color for light text colors
             break
         case LEGEND_DISPLAY_STYLE_FILL:
         default:
-            style.backgroundColor = legend.color
-            if (isColorBright(legend.color)) {
+            style.backgroundColor = legendColor
+            if (isColorBright(legendColor)) {
```

Comment:
@martinkrulltott this will break when `legendColor` is undefined, because `isColorBright` tries to parse it as a string.

Reply:
Should solved by re-adding the short-circuit from your other comment, but `getColorByValueFromLegendSet` could return an empty string if you prefer?

Reply:
I think that would also break unless we update the `isColorBright` implementation to accept non-color strings, so the contract for this function should be that it always accepts a color string.

Reply:
Ok, still with the short-circuit re-added it will never currently never reach `isColorBright`, right?